.evergreen/config.yml

```diff
@@ -1,48 +1,188 @@
+# Mark a failure as a system/bootstrap failure (purple box) rather then a task
+# failure by default.
+# Actual testing tasks are marked with `type: test`
+command_type: system
+
+# Protect ourselves against rogue test case, or curl gone wild, that runs forever
+exec_timeout_secs: 3600
+
+# When a task that used to pass starts to fail, go through all versions that may have been skipped
+# to detect when the task started failing
+stepback: true
+
 functions:
-  fetch source:
-    # Executes git clone
+  fetch-source:
+    # Executes git clone and applies the submitted patch, if any.
     - command: git.get_project
       params:
         directory: src
-    # Applies the submitted patch, if any
-    - command: git.apply_patch
 
-  exec script:
-    - command: shell.exec
-      type: test
+  create-expansions:
+    - command: subprocess.exec
       params:
         working_dir: src
-        shell: bash
-        add_to_path:
-          - .evergreen
+        binary: bash
+        include_expansions_in_env:
+          - is_patch
+          - version_id
+          - project
+        args:
+          - .evergreen/create-expansions.sh
+    - command: expansions.update
+      params:
+        file: src/expansion.yml
+
+  prepare-resources:
+    - command: subprocess.exec
+      params:
+        working_dir: src
+        binary: bash
+        include_expansions_in_env:
+          - DRIVERS_TOOLS
+        args:
+          - .evergreen/fetch-drivers-tools.sh
+
+  bootstrap-mongo-orchestration:
+    - command: subprocess.exec
+      params:
+        binary: sh
+        include_expansions_in_env:
+          - TOPOLOGY
+          - MONGODB_VERSION
+          - MONGODB_BINARIES
+        args:
+          - ${DRIVERS_TOOLS}/.evergreen/run-orchestration.sh
+    # run-orchestration generates expansion file with the MONGODB_URI for the cluster
+    - command: expansions.update
+      params:
+        file: mo-expansion.yml
+
+  stop-mongo-orchestration:
+    - command: shell.exec
+      params:
+        script: |
+          ${PREPARE_SHELL}
+          cd ${MONGO_ORCHESTRATION_HOME}
+          # source the mongo-orchestration virtualenv if it exists
+          if [ -f venv/bin/activate ]; then
+            . venv/bin/activate
+          elif [ -f venv/Scripts/activate ]; then
+            . venv/Scripts/activate
+          fi
+          mongo-orchestration stop || true
+
+  upload-test-results:
+    - command: attach.xunit_results
+      params:
+        file: src/build/test-results/test/TEST-*.xml
+
+  cleanup:
+    - command: shell.exec
+      params:
         script: |
-          ${script}
+          ${PREPARE_SHELL}
+          rm -rf $DRIVERS_TOOLS || true
```

Comment:
Here, `$DRIVERS_TOOLS` is an env var, that is set via the `PREPARE_SHELL` expansion. The `PREPARE_SHELL` expansion was set via the `create-expansions.sh` file. That same file appears to set a `DRIVERS_TOOLS` expansion. So, it seems we could write: 

`rm -rf ${DRIVERS_TOOLS} || true`

here to pick up that expansion. Then, we do not need to use `${PREPARE_SHELL}`.

Could we try this?

Also, it seems that in the other cases I see it used, we do not need to use `${PREPARE_SHELL}`, for roughly the same reason: all the values it sets can be provided as expansions to any script we run. Could we either document what it is doing, or create a ticket to do so?

1. If we do understand what PREPARE_SHELL does, we should consider leaving a comment documenting this, since this was not obvious, and it is not obvious to me what belongs there vs as an expansion.
2. If we do not understand, we could try to figure this out. If this would take too long, perhaps we should open a ticket? It seems strange to me to pass values around, sometimes via PREPARE_SHELL, sometimes via expansions, sometimes via sourced files, without a clear reason.

Reply:
`PREPARE_SHELL` is not only for duplicating env in current shell, it also pass them to subshell created, and set global shell behaviour as following:
```
    set -o errexit
    set -o xtrace
```
for this specific case, no subshell will be created, so we could use the above two to substitute for it usage. But using it consistently is a good practise. I could add some comment to it for I think its usage is obvious.

Reply:
I experimented removing the usage of `PREPARE_SHELL` and found it works. I think it is not needed currently so I removed its usage from `config.yml`.